cmd/tesla-control/commands.go

```diff
@@ -175,12 +193,12 @@ func checkReadiness(commandName string, havePrivateKey, haveOAuth, haveVIN bool)
 	return info, nil
 }
 
-func execute(ctx context.Context, acct *account.Account, car *vehicle.Vehicle, args []string) error {
+func execute(ctx context.Context, acct *account.Account, vehicle *vehicle.Vehicle, args []string) error {
 	if len(args) == 0 {
 		return errors.New("missing COMMAND")
 	}
 
-	info, err := checkReadiness(args[0], car != nil && car.PrivateKeyAvailable(), acct != nil, car != nil)
```

Comment:
Shouldn't re-use the same symbol for a package and a variable. This would break if there's a conflict in exported names/methods.

On the other hand, I agree that `car` isn't ideal here. I'd suggest just using `v` instead; this shorthand for a locally-scoped variable is pretty common in Go.

Reply:
Agreed with the non-awesomeness of shadowing the package with variables, but nothing will "break" though. Once shadowed, you just won't be able to use the underlying package while shadowed. For example:
https://go.dev/play/p/7VhEqRIMXYa
The main downside of shadowing is that you don't know what's what without really paying attention, and that can lead to confusion when reading the code.

I would recommend against `v` as a var name; this is common for receivers, but for one of many params in funcs can have have sizable scopes, it'd be better to have slightly more meaningful names.

In this particular case, adding an import alias like `veh` or `vehicl` or `pvehicle` or `vehiclep` or `vehipkg` or `somethingthatsnotvehicle` might work better, because it'll be very obvious that `vehicle veh.Vehicle` means `veh` is a package, and then the rest of the code reads nicely with `vehicle` as a subject to everything.

Reply:
Actually I think I'd just keep `car` in here.